lib/react.dart

```diff
@@ -616,9 +656,7 @@ class SyntheticWheelEvent extends SyntheticEvent {
 }
 
 /// Registers [componentFactory] on both client and server.
-Function registerComponent = (componentFactory, [skipMethods]) {
-  throw new Exception('setClientConfiguration must be called before registerComponent.');
-};
+ComponentRegistrar registerComponent = (ComponentFactory componentFactory, [Iterable<String> skipMethods]) => new _ReactComponentFactoryProxy();
```

Comment:
I think we went back and forth on this a couple times, but I think this should be:
```dart
ComponentRegistrar registerComponent = (ComponentFactory componentFactory, [Iterable<String> skipMethods]) {
  throw new Exception('setClientConfiguration must be called before registerComponent.');
}
```